### ISSUE TITLE
Implement mean_all & mean_keepdim

### ISSUE THREAD

**Issue by unrenormalizable** [huggingface/candle#1621]
mean_all / mean_keepdim are like sum_all / sum_keepdim but for mean. [numpy](https://numpy.org/doc/stable/reference/generated/numpy.mean.html) has this implemented. It is the torch.mean equivalent with the dim argument. It will be good to have this implemented.

The library I am working on [burn](https://github.com/tracel-ai/burn) - is adding this functionality but I have to implement it for candle in the burn codebase.
